ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -33,23 +34,43 @@ def __init__(self):
 
         rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
         rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
+        rospy.Subscriber('/traffic_waypoint', Int32, self.traffic_cb)
+        rospy.Subscriber('/obstacle_waypoint', Lane, self.obstacle_cb)
 
-        # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint
-
-        self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane,
-                                                   queue_size=1)
+        self.pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
 
         # TODO: Add other member variables you need below
+        self.world = None
+        self.ego = None
 
+        self.publish()
         rospy.spin()
 
+    def publish(self):
+        rate = rospy.Rate(5)
+        while not rospy.is_shutdown():
+            idx = self.find_next_waypoint()
+            if idx > -1:
+                rospy.loginfo("Next waypoint: {}".format(idx))
+                waypoints = self.world.waypoints + self.world.waypoints
+                waypoints = waypoints[idx:idx+LOOKAHEAD_WPS]
+                for i in range(LOOKAHEAD_WPS):
+                    self.set_waypoint_velocity(waypoints, i, 20)
```

Comment:
The velocity is already set to 40 mph from "waypoint_loader.py", and will automatically be set to 10 mph at the test place (waypoint_loader.launch and waypoint_loader_site.launch). Maybe we should reuse that information?

Reply:
I will remove those two lines and leave the velocity set by `waypoint_loader.py` for now then.